wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/document/EventLogScreen.kt

```diff
@@ -1,95 +1,233 @@
+fun EventLogScreen(
+    documentId: String,
+    documentModel: DocumentModel,
+    navController: NavHostController
+) {
     val eventInfos = remember {
         mutableStateListOf<EventInfo>().apply {
             addAll(documentModel.getEventInfos(documentId))
         }
     }
 
-    Box(
-        modifier = Modifier.fillMaxHeight()
+    ScreenWithAppBarAndBackButton(
+        title = stringResource(R.string.document_info_screen_menu_activities_log),
+        onBackButtonClick = { navController.popBackStack() }
     ) {
-        ScreenWithAppBarAndBackButton(
-            title = stringResource(R.string.document_info_screen_menu_activities_log),
-            onBackButtonClick = { onNavigate(WalletDestination.PopBackStack.route) }
-        ) {
-            if (eventInfos.isEmpty()) {
-                Row(
-                    modifier = Modifier.fillMaxWidth(),
-                    horizontalArrangement = Arrangement.Center,
-                ) {
-                    Text(
-                        modifier = Modifier.padding(8.dp),
-                        text = stringResource(R.string.document_info_screen_menu_activities_log_screen_empty),
-                        style = MaterialTheme.typography.titleLarge,
-                        textAlign = TextAlign.Center
-                    )
-                }
-            } else {
-                Column(modifier = Modifier.weight(1f)) {
+        if (eventInfos.isEmpty()) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Center,
+            ) {
+                Text(
+                    modifier = Modifier.padding(8.dp),
+                    text = stringResource(R.string.document_info_screen_menu_activities_log_screen_empty),
+                    style = MaterialTheme.typography.titleLarge,
+                    textAlign = TextAlign.Center
+                )
+            }
+        } else {
+            Column(
+                verticalArrangement = Arrangement.Top,  // Aligns content to the top
+                modifier = Modifier.weight(1f)
+            ) {
                 LazyColumn {
-                        items(eventInfos) { eventInfo ->
-                            Row(
-                                modifier = Modifier
-                                    .fillMaxWidth()
-                                    .padding(4.dp),
-                                verticalAlignment = Alignment.CenterVertically
-                            ) {
-                                KeyValuePairText(eventInfo.timestamp, eventInfo.requestedFields)
+                    items(eventInfos) { eventInfo ->
+                        Row(
+                            modifier = Modifier
+                                .fillMaxWidth()
+                                .padding(4.dp)
+                                .clickable {
+                                    // Pass documentId, requestedFields, and timestamp as arguments
```

Comment:
requestedFields isn't being passed...?

Reply:
No we can get the requested fields from the eventsInfo using the documentId

Reply:
Updated comment to be accurate.